types/parse/index.d.ts

```diff
@@ -825,7 +825,7 @@ namespace Parse {
     }
     interface UserConstructor extends ObjectStatic {
         new <T extends Attributes>(attributes: T): User<T>;
-        new(attributes?: Attributes): User;
+        new(): User;
```

Comment:
Why not letting create a User with attributes?

Reply:
This is the most natural way of using the SDK without using type params, I think that should be up to the user to decide.

Reply:
So perhaps I'm misinterpreting what you're asking - but the intent is for the overload on line 827 to cover the case where a User is created with initial attributes. In this way `new Parse.User({ attr: 'thing' })` and `new Parse.User<{ attr: string }>({ attr: 'thing' })` are both still allowed. The purpose of line 828 is to prevent `new User<{ attr: string }>()`, since that would create an object with a bad state (similar to the discussion for Parse.Object, TS would think `.get('attr')` was a string when it would actually be undefined).

These are the test cases that summarize those instances:
```ts
function testUser() {
    function testConstructor() {
        // $ExpectType User<Attributes>
        new Parse.User();

        // $ExpectType User<{ example: number; }>
        new Parse.User({ example: 100 });

        // $ExpectError
        new Parse.User<{ example: number }>();

        // $ExpectError
        new Parse.User<{ example: number }>({ example: 'hello' });
    }
}
```

Reply:
So, after some further thought, I think I realize what you were noticing. The irony here is that all the test cases that we care about work successfully whether `attributes?: Attributes` is a param or not. My instinct was not to include it purely for that reason - that it wasn't necessary. But I did notice that there is a subtle advantage to having it present, which is that it improves the intellisense for IDEs.

In other words, without the param present, someone might type `new Parse.User()` and in that moment, the second overload is being used and so hovering over `User` would not how `attributes` as a possible param. Of course, if they were to go ahead and pass a param anyway, then the first overload would be used, and hovering would correctly show the `attributes` param.

But *with* the param present in the second overload, hovering over `new Parse.User()` would show `attributes` as a possible param, which would be more helpful in guiding the user. And once they pass a parameter, the first overload will be used again as before.

So with that, I think that you're right that there is a subtle advantage there, so I added it back.

Reply:
Hey, actually I was about to comment and tell you that I got your point 😅. I missed the test this time as I was still chewing our previous conversation:
```
// $ExpectType User<{ example: number; }>
new Parse.User({ example: 100 });
```
but it actually works in any case. I'm glad at least a small advantage came out from this :)